lib/src/collection/multimap.dart

```diff
@@ -798,28 +798,28 @@ class _WrappedList<K, V> extends _WrappedIterable<K, V, List<V>>
   }
 
   @override
-  void shuffle([Random random]) {
+  void shuffle([Random? random]) {
     _syncDelegate();
     _delegate.shuffle(random);
   }
 
   @override
-  void sort([int compare(V a, V b)]) {
+  void sort([int compare(V a, V b)?]) {
     _syncDelegate();
     _delegate.sort(compare);
   }
 
   @override
-  List<V> sublist(int start, [int end]) {
+  List<V> sublist(int start, [int? end]) {
     _syncDelegate();
     return _delegate.sublist(start, end);
   }
 }
 
 class _WrappedSet<K, V> extends _WrappedIterable<K, V, Set<V>>
     implements Set<V> {
-  _WrappedSet(Map<K, Iterable<V>> map, K key, Iterable<V> delegate)
-      : super(map, key, delegate);
+  _WrappedSet(Map<K, Iterable<V>> map, K key, Set<V> delegate)
+      : super(map as Map<K, Set<V>>, key, delegate);
```

Comment:
Is this cast necessary? `SetMultimap<K, V> extends _BaseMultimap<K, V, Set<V>>`, which makes its `_map` field a `Map<K, Set<V>>`.

Reply:
Yep, since we're doing a narrowing cast.

Reply:
But why does the constructor take `Map<K, Iterable<V>> map` instead of `Map<K, Set<V>> map` in the first place?

Reply:
```dart
class _WrappedSet<K, V> extends _WrappedIterable<K, V, Set<V>>                  
    implements Set<V> {                                                         
  _WrappedSet(Map<K, Iterable<V>> map, K key, Set<V> delegate)                                                 
      : super(map, key, delegate);                                              
```

Reply:
Basically so we can avoid re-implementing everying twice.

Reply:
Discussed offline; amended the ctor generic collection type to use List<V>/Set<V> rather than Iterable<V> for C and we're all set.